lib/battle/CBattleInfoCallback.cpp

```diff
@@ -1308,34 +1309,34 @@ static bool isHexInFront(BattleHex hex, BattleHex testHex, BattleSide::Type side
 }
 
 //TODO: this should apply also to mechanics and cursor interface
-bool CBattleInfoCallback::isToReverse (const battle::Unit * attacker, const battle::Unit * defender) const
+bool CBattleInfoCallback::isToReverse(const battle::Unit * attacker, const battle::Unit * defender) const
 {
 	BattleHex attackerHex = attacker->getPosition();
 	BattleHex defenderHex = defender->getPosition();
 
 	if (attackerHex < 0 ) //turret
 		return false;
 
-	if (isHexInFront(attackerHex, defenderHex, BattleSide::Type(attacker->unitSide())))
+	if(isHexInFront(attackerHex, defenderHex, static_cast<BattleSide::Type>(attacker->unitSide())))

```

Comment:
@rilian-la-te (no need to change anything here)

@nullkiller here is an example of why static cast is preferable.

Currently, method `attacker->unitSide()` returns uint8_t, however that value is always interpreted as BattleSide::Type. This is indication of poor code. I can search code for `static_cast<BattleSide::Type>` and find all places where this happens (other than `unitSide`) and fix them if necessary.

Yes, previously you could search for `BattleSide::Type`, but it would show every usage of that type - function parameters, variable declarations, member declarations. Finding actual conversion would require checking all of that

Reply:
Okay) I will mark it as resolved, if you think than I do not need to do something, okay?